language_sentiment_text.rb

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# DO NOT EDIT! This is a generated sample ("Request",  "language_sentiment_text")
+
+# sample-metadata
+#   title: Analyzing Sentiment
+#   description: Analyzing Sentiment in a String
+#   bundle exec ruby samples/v1/language_sentiment_text.rb [--text_content "I am so happy and joyful."]
+
+require "google/cloud/language"
+
+# [START language_sentiment_text]
+
+# Analyzing Sentiment in a String
+#
+# @param text_content {String} The text content to analyze
+def sample_analyze_sentiment text_content
+  # Instantiate a client
+  language_client = Google::Cloud::Language.new version: :v1
+
+  # text_content = "I am so happy and joyful."
```

Comment:
For example arguments for the method, consider:
* Moving these to the top of the method
* Either expressing it as a default value (e.g. `text_content ||= "I am so happy and joyful."`) or including a comment that explains what you're doing (e.g. `# example value for the text_content argument`)

Reply:
Uncommenting a line to set the value when the variable is already provided by the method arguments is maddening. How is this not going to confuse users? If we have the method signatures, why not put them as default values for the method arguments?

```ruby
def sample_analyze_sentiment text_content: "I am so happy and joyful."
  # do stuff
end
```

Or, you know, don't have methods at all...

Reply:
I find the non-method version:

```ruby
# Analyzing Sentiment in a String
require "google/cloud/language"

# Instantiate a client
language_client = Google::Cloud::Language.new version: :v1

text_content = "I am so happy and joyful."

# Available types: PLAIN_TEXT, HTML
type = :PLAIN_TEXT
```

much easier to read than the method version:

```ruby
# Analyzing Sentiment in a String
#
# @param text_content {String} The text content to analyze
def sample_analyze_sentiment text_content
  # Instantiate a client
  language_client = Google::Cloud::Language.new version: :v1

  # text_content = "I am so happy and joyful."

  # Available types: PLAIN_TEXT, HTML
  type = :PLAIN_TEXT
```

Reply:
@frankyn @TheRoyalTnetennba Do we have a strong/strict lean towards using methods -vs- no using methods?

[`ruby-docs-samples/CONTRIBUTING`](https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/master/CONTRIBUTING.md#code-samples) shows both, but I think no method for these types of snippets.

I'd like to ship a version without method signatures and see how that looks! Particularly in response to these truths:

> Uncommenting a line to set the value when the variable is already provided by the method arguments is maddening. How is this not going to confuse users?

Reply:
Let's try no method signature. I agree it's cleaner.